crates/rolldown/tests/fixtures/deconflict/wrapped_esm_default_function/artifacts.snap

```diff
@@ -31,9 +31,9 @@ var bar_default = {foo: foo$1};
 //#endregion
 //#region main.js
 const a = 2;
-const { foo } = bar_default;
-assert.strictEqual(typeof foo, 'function');
-(init_foo(), __toCommonJS(foo_ns));
+const { foo: foo } = bar_default;
```

Comment:
This change is also not expected, https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBA3jAZiEMC-MC8MBGBDAJwCgig&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=Electron-1.8%252CNode-10.13&version=7.24.10&externalPlugins=&assumptions=%7B%7D

Reply:
Caused by https://github.com/oxc-project/oxc/pull/4319. I will create an issue to track this